### PR TITLE
Response Planner

### PR DESCRIPTION
**Pull request by garunagiri**
Lots of changes here. I think every big piece is addressed by a test.

### CODE REVIEW

**Comment by garunagiri** [comment]
The severity stuff requires https://github.com/mitre/caldera/pull/1722

**Comment by wbooth** [comment]
run through operation
read through top to bottom then submit for review

File: app/response_planner.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+        await self.planning_svc.exhaust_bucket(self, ['detection'], self.operation, batch=True)
+        self.next_bucket = await self.planning_svc.default_next_bucket('detection', self.state_machine)
+
+    async def hunt(self):
+        await self._do_reactive_bucket(bucket='hunt')
+        self.next_bucket = await self.planning_svc.default_next_bucket('hunt', self.state_machine)
+
+    async def response(self):
+        await self._do_reactive_bucket(bucket='response')
+        self.next_bucket = await self.planning_svc.default_next_bucket('response', self.state_machine)
```
Comment (wbooth) [inline app/response_planner.py:54]: are these the same function?

File: app/response_planner.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        Get all the links in the operation's chain that contain the fact in at least one relationship.
+        """
+        links_with_fact = []
+        for link in self.operation.chain if paw is None else [lnk for lnk in self.operation.chain if lnk.paw == paw]:
+            if any(self._fact_in_relationship(fact, rel) for rel in link.relationships):
+                links_with_fact.append(link)
+        return links_with_fact
+
+    def _fact_in_relationship(self, fact, relationship):
+        for f in [relationship.source, relationship.target]:
```
Comment (wbooth) [inline app/response_planner.py:138]: single letter variables can be ambiguous might want to just be explicit here

**Comment by garunagiri** [comment]
Read through the code, and tested it through normal use with two agents.

**Comment by garunagiri** [comment]
Setting repeatable to True for some abilities would cause issues when using the default planners, but would work correctly with this planner. What would be the appropriate step to take here?

**Comment by wbooth** [comment]
discussed options for re-write

**Pull request closed without merging**